urlfd018a75f7_highlanderkev_StockExVirtuoso_tgz-pJ8-driver_Phase4J8/src/driver/Phase1.java

```diff
@@ -0,0 +1,244 @@
+ * if you want to put this class in a different package, you need to change the
+ * package declaration below to match the desired package name.
+ */
+package driver;
+
+import java.util.ArrayList;
+import price.Price;
+import price.PriceFactory;
+import price.utils.InvalidPriceOperation;
+
+
+
+public class Phase1 {
+
+    private static ArrayList<Price> testPriceHolder = new ArrayList<>();
+
+    public static void main(String[] args) throws Exception {
+        makeSomeTestPriceObjects();
+        verifyTestPriceValues();
+        verifyMathematicalOperations();
+        verifyBooleanChecks();
+        verifyComparisons();
+        verifyFlyweight();
+    }
+
+    private static void makeSomeTestPriceObjects() throws Exception {
+        System.out.println("Creating some Test Price Objects.");
+        testPriceHolder.add(PriceFactory.makeLimitPrice("10.50"));
+        testPriceHolder.add(PriceFactory.makeLimitPrice("$1400.99"));
+        testPriceHolder.add(PriceFactory.makeLimitPrice("$-51.52"));
+        testPriceHolder.add(PriceFactory.makeLimitPrice(".49"));
+        testPriceHolder.add(PriceFactory.makeLimitPrice("-0.89"));
+        testPriceHolder.add(PriceFactory.makeLimitPrice("12"));
+        testPriceHolder.add(PriceFactory.makeLimitPrice("90."));
+        testPriceHolder.add(PriceFactory.makeLimitPrice("14.5"));
+        testPriceHolder.add(PriceFactory.makeMarketPrice());
+    }
+
+    private static void verifyTestPriceValues() {
+        System.out.println("Verifying the Values in Your Test Price Objects:");
+        String format = "%-9s --> %9s : %s%n";
+        System.out.format(format, "$10.50", testPriceHolder.get(0), testPriceHolder.get(0).toString().equals("$10.50") ? "CORRECT" : "ERROR");
+        System.out.format(format, "$1,400.99", testPriceHolder.get(1), testPriceHolder.get(1).toString().equals("$1,400.99") ? "CORRECT" : "ERROR");
+        System.out.format(format, "$-51.52", testPriceHolder.get(2), testPriceHolder.get(2).toString().equals("$-51.52") ? "CORRECT" : "ERROR");
+        System.out.format(format, "$0.49", testPriceHolder.get(3), testPriceHolder.get(3).toString().equals("$0.49") ? "CORRECT" : "ERROR");
+        System.out.format(format, "$-0.89", testPriceHolder.get(4), testPriceHolder.get(4).toString().equals("$-0.89") ? "CORRECT" : "ERROR");
+        System.out.format(format, "$12.00", testPriceHolder.get(5), testPriceHolder.get(5).toString().equals("$12.00") ? "CORRECT" : "ERROR");
+        System.out.format(format, "$90.00", testPriceHolder.get(6), testPriceHolder.get(6).toString().equals("$90.00") ? "CORRECT" : "ERROR");
+        System.out.format(format, "$14.50", testPriceHolder.get(7), testPriceHolder.get(7).toString().equals("$14.50") ? "CORRECT" : "ERROR");
+        System.out.format(format, "MKT", testPriceHolder.get(8), testPriceHolder.get(8).toString().equals("MKT") ? "CORRECT" : "ERROR");
+        System.out.println();
+    }
+
+    private static void verifyMathematicalOperations() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 60-114 into a separate method.